salt/modules/boto_secgroup.py

```diff
@@ -221,15 +242,22 @@ def get_group_id(name, vpc_id=None, region=None, key=None, keyid=None, profile=N
     '''
     conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
 
-    group = _get_group(conn, name, vpc_id, region)
+    if not vpc_id and vpc_name:
```

Comment:
Hmm, I tried that in one of my earlier iterations and hit some snag which made me back it out.  I do see the benefit though so I'll take another stab at making it work...

Thanks for your input BTW, I really appreciate getting good criticism :)

Hmm I'm also getting pylint errors again so it's back to the trenches for me...